### PR TITLE
Add more compiler version info to INSTALL.rst

### PR DESCRIPTION
**Pull request by aradi**
In order to make working compiler version more explicit.

### CODE REVIEW

**Review by awvwgk — APPROVED** [review]
Looks good to me.

File: INSTALL.rst [hunk 1]
```diff
@@ -27,6 +27,34 @@ In order to compile DFTB+, you need the following software components:
 * Python (version >= 3.2) for the source preprocessor
 
 
+Fortran compiler
+----------------
+
+The following Fortran compilers are known to build DFTB+:
```
Comment (bhourahine) [inline INSTALL.rst:33]:
```suggestion
The following Fortran compilers are known to build DFTB+ correctly:
```

File: INSTALL.rst [hunk 2]
```diff
@@ -27,6 +27,34 @@ In order to compile DFTB+, you need the following software components:
...
+
+* NAG (when built with OpenMP support, unsupported OpenMP 4.0 constructs, last
+  tested version: 7.1)
+
+* NVIDIA (internal compiler error & unsupported OpenMP 4.0 constructs, last
+  tested version: 22.3)
+
+
+Older versions of the compilers above are likely to fail due to missing Fortran
+features and/or compiler bugs, please use the versions listed above. Compilers
```
Comment (bhourahine) [inline INSTALL.rst:53]:
```suggestion
features and/or compiler bugs. Compilers
```
Think the "please use" part is implied.

File: INSTALL.rst [hunk 3]
```diff
@@ -27,6 +27,34 @@ In order to compile DFTB+, you need the following software components:
...
+by other vendors may work, but had not been tested extensively (see also
+`Tested build environments <#tested-build-environments>`_).
```
Comment (bhourahine) [inline INSTALL.rst:55]:
```suggestion
`Tested build environments <#tested-build-environments>`_ and `Testing DFTB+ <#testing-dftb+>`_).
```
(haven't tested the suggested link in place)

File: INSTALL.rst [hunk 4]
```diff
@@ -27,6 +27,34 @@ In order to compile DFTB+, you need the following software components:
...
+* NAG (when built with OpenMP support, unsupported OpenMP 4.0 constructs, last
+  tested version: 7.1)
+
+* NVIDIA (internal compiler error & unsupported OpenMP 4.0 constructs, last
+  tested version: 22.3)
+
+
+Older versions of the compilers above are likely to fail due to missing Fortran
+features and/or compiler bugs, please use the versions listed above. Compilers
+by other vendors may work, but had not been tested extensively (see also
```
Comment (bhourahine) [inline INSTALL.rst:54]:
```suggestion
by other vendors may work, but have not been tested extensively (see also
```

File: INSTALL.rst [hunk 5]
```diff
@@ -27,6 +27,34 @@ In order to compile DFTB+, you need the following software components:
...
 
 
+Fortran compiler
+----------------
+
+The following Fortran compilers are known to build DFTB+:
+
+* GNU >= 9
+
+* Intel >= 19 or >= 2020 (when built with the tblite library, tblite is known to
```
Comment (bhourahine) [inline INSTALL.rst:37]:
```suggestion
* Intel >= 19 or >= 2020 (when built to include the tblite library, as tblite is known to
```

**Review by bhourahine — APPROVED** [review]
Ready to merge, but see suggestions.

**Pull request merged**